app/scripts/controllers/network/network-controller.test.js

```diff
@@ -5393,42 +6751,44 @@ describe('NetworkController', () => {
               },
             },
           },
-          async ({ controller, network: network1 }) => {
-            network1.mockEssentialRpcCalls({
-              net_version: {
-                response: {
-                  result: '42',
-                },
-              },
-            });
-            const network2 = network1.with({
+          async ({ controller, network: previousNetwork }) => {
+            const currentNetwork = new NetworkCommunications({
               networkClientType: 'infura',
               networkClientOptions: {
                 infuraNetwork: 'goerli',
               },
             });
-            network2.mockEssentialRpcCalls();
+            currentNetwork.mockEssentialRpcCalls({
+              // This results in a successful call to eth_getBlockByNumber
+              // implicitly
+              latestBlock: BLOCK,
+            });
+            previousNetwork.mockEssentialRpcCalls({
+              net_version: {
+                response: UNSUCCESSFUL_JSON_RPC_RESPONSE,
+              },
+            });
 
             await waitForLookupNetworkToComplete({
               controller,
               operation: () => {
                 controller.setProviderType('goerli');
               },
             });
-            expect(controller.store.getState().network).toBe('5');
+            expect(controller.store.getState().networkStatus).toBe('available');
 
             await waitForLookupNetworkToComplete({
               controller,
               operation: () => {
                 controller.rollbackToPreviousProvider();
               },
             });
-            expect(controller.store.getState().network).toBe('42');
+            expect(controller.store.getState().networkStatus).toBe('unknown');
```

Comment:
Hopefully the previous network is still available, but the code goes through the same process that it would if you were switching to that network for the first time. I guess I don't really know what the proper behavior should be here, but I'm just testing what the code does.

Reply:
I guess it would seem strange to call `rollbackToPreviousProvider` on a network where the status is `available`? I guess I was thinking the point of this method is to rollback when the network isn't available.

Reply:
Ah, I see what you mean about it being opposite from reality. Sorry didn't even think about when writing this test 😅 Fixed in [d2b5d78](https://github.com/MetaMask/metamask-extension/pull/17556/commits/d2b5d789314cc895271c3f3b55ea881b83de5ab4).